### ISSUE TITLE
psvelo _always_ scales vector attributes

### ISSUE THREAD

**Issue by anne-glerum** [GenericMappingTools/gmt#2035]
**Description of the problem**
Using gmt 5.4.5 on my Mac, I can't get the vectors I plot with psvelo to have a fixed head and body width. According to the manual, this should be the default behaviour, however, the vector attributes are scaled with vector length whether or not I use the -A+n option or not. Am I missing something or is this really unexpected behaviour?

**Full script that generated the error**

```
#!/bin/bash 

rm test.ps
rm test.pdf

vscale=600

#### should be with fixed vector attributes ####
gmt psvelo -JX3i/4i -R0/2100/0/2700  -Bxyf300 -Bx0 -By0 -P test.csv -A+pthicker+ea -Se$vscale/0/8 -V -G42.591/31.591/11.591 -W0.5,42.591/31.591/11.591 -N -K > test.ps 
#### should be with scaled vector attributes ####
gmt psvelo -J -R -P test.csv -A+pthicker+ea+n0.05 -Se$vscale/0/8 -V -Ggreen -W0.1,green -N -O >> test.ps

#### convert ps to pdf ####
gmt psconvert test.ps -Tf -A -V  

#### open pdf for inspection ####
open test.pdf
```

**Input data**
````
300 100 -0.00228787 -6.12987e-08 0 0 0 
300 600 -0.0023007 -4.15962e-06 0 0 0 
300 1100 -0.00226568 -1.15024e-06 0 0 0 
300 1600 -0.00203787 -5.35509e-05 0 0 0 
300 2100 -0.00204038 -0.000131858 0 0 0 
300 2600 -0.00214157 -2.72306e-05 0 0 0 
800 100 -0.00203193 4.42606e-06 0 0 0 
800 600 -0.00190225 4.36551e-05 0 0 0 
800 1100 0.000459517 -0.000462841 0 0 0 
800 1600 -0.000670736 -0.000364103 0 0 0 
800 2100 -0.00151165 -0.000115605 0 0 0 
800 2600 -0.00184433 -1.25489e-05 0 0 0 
1300 100 0.0018441 1.27042e-05 0 0 0 
1300 600 0.00151107 0.000115925 0 0 0 
1300 1100 0.000670381 0.000364669 0 0 0 
1300 1600 -0.000459696 0.000463376 0 0 0 
1300 2100 0.00190218 -4.35958e-05 0 0 0 
1300 2600 0.00203209 -4.40132e-06 0 0 0 
1800 100 0.00214217 2.72031e-05 0 0 0 
1800 600 0.00204121 0.00013186 0 0 0 
1800 1100 0.00203786 5.37396e-05 0 0 0 
1800 1600 0.00226553 1.46369e-06 0 0 0 
1800 2100 0.00229981 4.13721e-06 0 0 0 
1800 2600 0.00228223 -5.6886e-07 0 0 0
````

**Full error message**
There is no error message.

**Actual outcome**

[test.pdf](https://github.com/GenericMappingTools/gmt/files/3836541/test.pdf)

**Expected outcome**

I expect the black set of vectors to have a body and head width that is invariant to vector length. However, the green and the black set are equal.

**System information**

* Operating system: MacOS Majove 10.14.5
* GMT version (`gmt --version`): 5.4.5

**Comment by PaulWessel** [comment]
So there are some ancient code in psvelo that looks a bit strange to me.  This is the section that decides on scaling the attributes:

```
if (hypot (plot_x-plot_vx, plot_y-plot_vy) <= 1.5 * Ctrl->A.S.v.h_length) {
	hl = hypot (plot_x-plot_vx,plot_y-plot_vy) * 0.6;
	hw = hl * Ctrl->A.S.v.h_width/Ctrl->A.S.v.h_length;
	vw = hl * Ctrl->A.S.v.v_width/Ctrl->A.S.v.h_length;
	if (vw < 2.0/PSL_DOTS_PER_INCH) vw = 2./PSL_DOTS_PER_INCH;
}
```

First, there is a check that the length of the vector is less than 1.5 the head length. However, in psxy and presumably here we want **+n**_size_ to dictate when things should start shrinking.  So **+n** is not used.  Since all the actual vectors plotted in the user's example are longer than 0.125i [Default head size] then no shrinking takes place.  There is also the oddness of PSL_DOTS_PER_INCH to ensure a vector is not vanishing.  This dates back when that DPI was 300, now it is 1200.

In psxy we instead do this:

```
s = (length < S.v.v_norm) ? length / S.v.v_norm : 1.0;
dim[0] = x_2, dim[1] = y_2;
dim[2] = s * S.v.v_width, dim[3] = s * S.v.h_length, dim[4] = s * S.v.h_width;
```

i.e., attributes are scaled if the vector length is less than the specified **+n**_length_.

I suggest we ensure psvelo behaves in the same manner.

**Comment by PaulWessel** [comment]
What say you, @joa-quim and @seisman ?

**Comment by seisman** [comment]
I agree.

**Comment by PaulWessel** [comment]
What says the elusive Portuguese @joa-quim ?

**Comment by joa-quim** [comment]
I say that I keep having visits and that we agree to make it doing the same thing as psxy.

**Comment by anne-glerum** [comment]
Thank you for the quick reply!! 
I just installed the developer version to try #2046 and I'm slightly confused now. Without the +n option, vector heads and thicknesses are still scaled with size and with the +n I frankly don't see a difference. The test added in #2046 shows exactly the same vectors (black and orange) on my machine. Is that meant to be?

**Comment by PaulWessel** [comment]
The result of the new test is given below - clearly the smaller orange vectors are shrinking.  In your original example, you used **+n**0.05 which means no shrinking will take place until the length of a vector is < half a mm.  None of your vectors were even remotely close to being that small, hence no change.  As you see, the test is using **+n**0.5i and the orange vectors shorter than that have their attributes scaled accordingly.

![geodesy_06](https://user-images.githubusercontent.com/26473567/68784550-d30ed800-05e0-11ea-926c-8928405f20f5.jpg)

**Comment by seisman** [comment]
@PaulWessel The geodesy_06 test fails in our CI, possibly because the original PS was generated with SI units, while the gm test uses US units.

**Comment by PaulWessel** [comment]
Ay, I guess it is the unitless -Se600?  Will update.

**Comment by anne-glerum** [comment]
@PaulWessel That figure looks great, thanks. Unfortunately, mine looks like this:
[test.pdf](https://github.com/GenericMappingTools/gmt/files/3845988/test.pdf)
based on this script based on the test:

    #!/bin/bash 

    rm test.pdf

    cat &lt;&lt; EOF &gt; test2.csv
    300 100 -0.00228787 -6.12987e-08 0 0 0 
    300 600 -0.0023007 -4.15962e-06 0 0 0 
    300 1100 -0.00226568 -1.15024e-06 0 0 0 
    300 1600 -0.00203787 -5.35509e-05 0 0 0 
    300 2100 -0.00204038 -0.000131858 0 0 0 
    300 2600 -0.00214157 -2.72306e-05 0 0 0 
    800 100 -0.00203193 4.42606e-06 0 0 0 
    800 600 -0.00190225 4.36551e-05 0 0 0 
    800 1100 0.000459517 -0.000462841 0 0 0 
    800 1600 -0.000670736 -0.000364103 0 0 0 
    800 2100 -0.00151165 -0.000115605 0 0 0 
    800 2600 -0.00184433 -1.25489e-05 0 0 0 
    1300 100 0.0018441 1.27042e-05 0 0 0 
    1300 600 0.00151107 0.000115925 0 0 0 
    1300 1100 0.000670381 0.000364669 0 0 0 
    1300 1600 -0.000459696 0.000463376 0 0 0 
    1300 2100 0.00190218 -4.35958e-05 0 0 0 
    1300 2600 0.00203209 -4.40132e-06 0 0 0 
    1800 100 0.00214217 2.72031e-05 0 0 0 
    1800 600 0.00204121 0.00013186 0 0 0 
    1800 1100 0.00203786 5.37396e-05 0 0 0 
    1800 1600 0.00226553 1.46369e-06 0 0 0 
    1800 2100 0.00229981 4.13721e-06 0 0 0 
    1800 2600 0.00228223 -5.6886e-07 0 0 0
    EOF

    #### plot
    gmt begin test pdf 
    gmt --version
    gmt velo -JX6.5i/9i -R0/2100/0/2700 -B test2.csv -A+pthicker+ea -Se600/0/8 -Gblack -W0.1p,black 
    gmt velo test2.csv -A+pthicker+ea+n0.5i -Se600/0/8 -Gorange -W0.1p,orange 

    #### convert ps to pdf ####
    gmt psconvert -Tf -A -V -Ftest.pdf

    gmt end 

    #### open pdf for inspection ####
    open test.pdf

**Comment by PaulWessel** [comment]
That can only (?) mean you are not really running the latest 6.0.x branch  What does gmt --version return?  BTW, you can scrap those psconvert/open calls and just say gmt end show.

**Comment by PaulWessel** [comment]
Make sure you select the 6.0 branch since #2046 was merged yesterday.

**Comment by seisman** [comment]
I believe the issue was fixed in #2046, and psvelo should work as expected in both 6.0 and master branches. Closing. Feel free to reopen it if you still have the same issue.

**Comment by anne-glerum** [comment]
> That can only (?) mean you are not really running the latest 6.0.x branch What does gmt --version return? 
My bad! I installed the dev branch 6.1.0_2131a19_2019.11.12, but that didn't include your fix yet. Now (with 6.1.0_4ad8dd5_2019.11.17) it works, thank you for the quick solution!
